### DISCUSSION TITLE
why "use qtTrId() for translations"

### DISCUSSION THREAD

**Discussion by cezaryece** [cutelyst/cutelyst#432] [General]
As a programmer, I'm surprised, even though I've seen such ideas before. In my opinion it makes the code less readable...

**Comment by buschmann23** [comment]
One big advantage of using id based translations is that the id does not change if the source string changes. If you have for example a typo in the source string and fix it, the translations for other languages still work as the id has not changed. It is also easier to change the source string for same ids as the source string is only there one time.

You are still free to use translate() method instead of qtTrId() in your own application. Both is supported but you should not mix it in the same translation file.

**Reply by cezaryece** [reply]
quite logical
thx
